src/main/java/com/scally/serverutils/executors/SlabsCommandExecutor.java

```diff
@@ -106,6 +115,40 @@ public boolean onCommand(@NotNull CommandSender commandSender, @NotNull Command
         return true;
 
     }
+    public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
+        if (!(sender instanceof Player)) {
+            return Collections.EMPTY_LIST;
+        }
+        Player player = (Player) sender;
+        Block targ = player.getTargetBlock(Set.of(Material.AIR, Material.CAVE_AIR, Material.VOID_AIR, Material.WATER), 5);
+        switch(args.length) {
+            case 1, 4:
+                return List.of(targ.getX() + "", targ.getX() + " " + targ.getY(), targ.getX() + " " + targ.getY() + " " + targ.getZ() );
+            case 2, 5:
+                return List.of(targ.getY() + "", targ.getY() + " " + targ.getZ() );
+            case 3, 6:
+                return List.of(targ.getZ() + "" );
+            case 7, 8:
+                final String prefix = args[args.length-1];
+
+                String[] parts = prefix.split(",");
+                final List<String> slabs = Arrays.asList(parts);
+                /*for(int i = 0; i < slabs.size(); i++) {
+                    sender.sendMessage(slabs.get(i));
+                }*/
+
+                return Tag.SLABS.getValues()
+                        .stream()
+                        .map(Material::toString)
+                        .map(String::toLowerCase)
+                        .filter(s -> s.startsWith(slabs.get(slabs.size()-1)))
```

Comment:
If I understand this correctly, I think the problem is that you are not concatenating the previous value of the arg to the generated list. So if the current value of the arg is `oak_slab,b`, then you are properly getting the list of the slabs that start with `b` but you'll also need to make sure that each of those values has `oak_slab,` in front of it.

So instead of returning a list of `["birch_slab", "blackstone_slab", ...]` you'll want to return `["oak_slab,birch_slab", "oak_slab,blackstone_slab", ...]`. Lemme know if that doesn't make sense